vcd/resource_vcd_nsxt_edgegateway.go

```diff
@@ -0,0 +1,414 @@
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error updating NSX-T edge gateway type: %s", err))
+	}
+
+	updatedEdge.ID = edge.EdgeGateway.ID
+	edge.EdgeGateway = updatedEdge
+
+	_, err = edge.Update(edge.EdgeGateway)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error updating edge gateway with ID '%s': %s", d.Id(), err))
+	}
+
+	return resourceVcdNsxtEdgeGatewayRead(ctx, d, meta)
+}
+
+// resourceVcdNsxtEdgeGatewayRead
+func resourceVcdNsxtEdgeGatewayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("[TRACE] NSX-T edge gateway read initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error retrieving VDC: %s", err))
+	}
+
+	edge, err := vdc.GetNsxtEdgeGatewayById(d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T edge gateway: %s", err))
+	}
+
+	err = setNsxtEdgeGatewayData(edge.EdgeGateway, d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading NSX-T edge gateway data: %s", err))
+	}
+	return nil
+}
+
+// resourceVcdNsxtEdgeGatewayDelete
+func resourceVcdNsxtEdgeGatewayDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("[TRACE] edge gateway deletion initiated")
+
+	vcdClient := meta.(*VCDClient)
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error retrieving VDC: %s", err))
+	}
+
+	edge, err := vdc.GetNsxtEdgeGatewayById(d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T edge gateway: %s", err))
+	}
+
+	err = edge.Delete()
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error deleting NSX-T edge gateway: %s", err))
+	}
+
+	return nil
+}
+
+// resourceVcdNsxtEdgeGatewayImport
+func resourceVcdNsxtEdgeGatewayImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[TRACE] NSX-T edge gateway import initiated")
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 3 {
+		return nil, fmt.Errorf("resource name must be specified as org-name.vdc-name.nsxt-edge-gw-name")
+	}
+	orgName, vdcName, edgeName := resourceURI[0], resourceURI[1], resourceURI[2]
+
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find org %s: %s", vdcName, err)
+	}
+
+	edge, err := vdc.GetNsxtEdgeGatewayByName(edgeName)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve NSX-T edge gateway with ID '%s': %s", d.Id(), err)
+	}
+
+	_ = d.Set("org", orgName)
+	_ = d.Set("vdc", vdcName)
+
+	d.SetId(edge.EdgeGateway.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+// getNsxtEdgeGatewayType
+func getNsxtEdgeGatewayType(d *schema.ResourceData, vdc *govcd.Vdc) (*types.OpenAPIEdgeGateway, error) {
```

Comment:
```suggestion
func transformToNsxtEdgeGatewayType(d *schema.ResourceData, vdc *govcd.Vdc) (*types.OpenAPIEdgeGateway, error) {
```

Reply:
Could change, but we never used such wording for converting "schema" to "type":
```
$ grep -REin "^func.*get.*type.*" *.go | wc -l
      51
$ grep -REin "^func.*transform.*" *.go | wc -l
       0
```

Reply:
well I am bias guy, for me `get` means try to search/query/fetch/dig something - not convert from input to output type. Can be different naming, yet if it's ok with everyone, I can live with it.

Reply:
I personally don't like `transformTo`